src/components/views/beacon/RoomCallBanner.tsx

```diff
@@ -0,0 +1,180 @@
+import { Action } from "../../../dispatcher/actions";
+import { Call, ConnectionState, ElementCall } from "../../../models/Call";
+import { useCall } from "../../../hooks/useCall";
+import { RoomViewStore } from "../../../stores/RoomViewStore";
+import { useEventEmitterState } from "../../../hooks/useEventEmitter";
+import {
+    OwnBeaconStore,
+    OwnBeaconStoreEvent,
+} from "../../../stores/OwnBeaconStore";
+import { CallDurationFromEvent } from "../voip/CallDuration";
+import { MockedCall } from "../../../../test/test-utils/call";
+
+interface RoomCallBannerProps {
+    roomId: Room["roomId"];
+    call: Call;
+}
+
+const RoomCallBannerInner: React.FC<RoomCallBannerProps> = ({
+    roomId,
+    call,
+}) => {
+    let callEvent: MatrixEvent = null;
+
+    if (!!(call as ElementCall).groupCall) {
+        callEvent = (call as ElementCall).groupCall;
+    }
+    if (!!(call as MockedCall).event) {
+        callEvent = (call as MockedCall).event;
+    }
+
+    const connect = useCallback(
+        (ev: ButtonEvent) => {
+            ev.preventDefault();
+            defaultDispatcher.dispatch<ViewRoomPayload>({
+                action: Action.ViewRoom,
+                room_id: roomId,
+                view_call: true,
+                metricsTrigger: undefined,
+            });
+        },
+        [roomId],
+    );
+
+    const disconnect = useCallback(
+        (ev: ButtonEvent) => {
+            ev.preventDefault();
+            call?.disconnect();
+        },
+        [call],
+    );
+
+    const onClick = () => {
+        dispatcher.dispatch<ViewRoomPayload>({
+            action: Action.ViewRoom,
+            room_id: roomId,
+            metricsTrigger: undefined,
+            event_id: callEvent.getId(),
+            scroll_into_view: true,
+            highlighted: true,
+        });
+    };
+
+    let [buttonText, buttonKind, onButtonClick] = [null, null, null, null];
+
+    switch (call.connectionState) {
+        case ConnectionState.Disconnected:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Join"),
+                "primary",
+                connect,
+            ];
+            break;
+        case ConnectionState.Connecting:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Join"),
+                "primary",
+                null,
+            ];
+            break;
+        case ConnectionState.Connected:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Leave"),
+                "danger",
+                disconnect,
+            ];
+            break;
+        case ConnectionState.Disconnecting:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Leave"),
+                "danger",
+                null,
+            ];
+            break;
+    }
+    if (!call) return <Fragment />;
+
+    return (
+        <div
+            className="mx_RoomLiveShareWarning mx_RoomCallBanner"
```

Comment:
I'd suggest renaming `mx_RoomLiveShareWarning` to something different if the intent is to share styles between the two banner types

Reply:
my 2c: styles shouldn't be shared, if necessary a reusable component should be abstracted out. In the short term I think it's better to duplicate the styles in `mx_RoomCallBanner`